include/readdy/kernel/singlecpu/actions/SCPUMdgfrdIntegrator.h

```diff
@@ -0,0 +1,165 @@
+ *     copyright notice, this list of conditions and the following  *
+ *     disclaimer in the documentation and/or other materials       *
+ *     provided with the distribution.                              *
+ *  3. Neither the name of the copyright holder nor the names of    *
+ *     its contributors may be used to endorse or promote products  *
+ *     derived from this software without specific                  *
+ *     prior written permission.                                    *
+ *                                                                  *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND           *
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,      *
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF         *
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE         *
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR            *
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,     *
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,         *
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; *
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER *
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,      *
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    *
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF      *
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.                       *
+ ********************************************************************/
+
+
+/**
+ * @todo
+ *
+ * @file SCPUMdgfrdIntegrator.h
+ * @brief << brief description >>
+ * @author chrisfroe
+ * @author luigisbailo
+ * @date 12.02.19
+ */
+#pragma once
+#include <readdy/model/actions/Actions.h>
+#include <readdy/kernel/singlecpu/SCPUKernel.h>
+#include <readdy/common/boundary_condition_operations.h>
+
+namespace readdy {
+namespace kernel {
+namespace scpu {
+
+namespace actions {
+
+struct MdgfrdParticleData {
+    scalar exitTime = 0;
+    scalar constructionTime = 0;
+    scalar domainSize = 0;
+
+    bool GFpropagation = false;
+};
+
+struct MdgfrdSpeciesData {
+    scalar minDomainSize = 0;
+    scalar maxDomainSize = 0;
+};
+
+
+class SCPUMdgfrdIntegrator : public readdy::model::actions::MdgfrdIntegrator {
+public:
+
+    SCPUMdgfrdIntegrator(SCPUKernel *kernel, scalar timeStep)
+            : readdy::model::actions::MdgfrdIntegrator(timeStep), kernel(kernel) {
+        const auto &context = kernel->context();
+        const auto &particlesTypeIds = context.particleTypes().typesFlat();
+        // set up species data
+        for (const auto& speciesI : particlesTypeIds) {
+            const auto &infoI = context.particleTypes().infoOf(speciesI);
+            MdgfrdSpeciesData speciesDataI;
+            // fixme: 4 is just a guess
+            speciesDataI.minDomainSize = 4. * std::sqrt(infoI.diffusionConstant * timeStep);
```

Comment:
This will be fixed, when we actually implement it. Currently the ActionFactory does not allow to construct this action. We can withdraw everything in this header and put it in the next PR if you'd prefer that.

Reply:
No that's fine by me, I realized that it was a draft later and forgot to delete the remark